app/dal/CassandraCluster.scala

```diff
@@ -25,11 +28,22 @@ class CassandraCluster @Inject()(conf: CassandraConfig, appLifecycle: Applicatio
 }
 
 @Singleton
-class CassandraConfig @Inject()(config: Configuration) {
-  val keySpace = config.getString("devgym.db.cassandra.keyspace").get
-  val port = config.getInt("devgym.db.cassandra.port").get
+class CassandraConfig @Inject()(configuration: Configuration) {
+  val config: Config = configuration.underlying
+
+  val keySpace = config.getString("devgym.db.cassandra.keyspace")
+  val port = config.getInt("devgym.db.cassandra.port")
   val hosts: Seq[String] = {
-    import scala.collection.JavaConversions._
-    config.getStringList("devgym.db.cassandra.hosts").get
+    def hosts: Seq[String] = {
+      import scala.collection.JavaConversions._
+      config.getStringList("devgym.db.cassandra.hosts")
+    }
+    def docker: Option[Boolean] = configuration.getBoolean("devgym.db.cassandra.docker")
+    def ip() = docker match {
+      case Some(true) => Try(Seq("docker-machine", "ip", "default").!!.trim).toOption
+      case _ => None
+    }
+
+    ip().fold(hosts) { ip => Seq(ip) }
```

Comment:
Fold is common functional operation.

Here is short protocol:
Lets calculate hosts!
Try to figure out ip somehow. If failed use hosts from config else use discovered ip.
Now about ip: if we can try docker, let's try it.